benchmarks/lockhammer/src/lockhammer.c

```diff
@@ -254,19 +263,24 @@ void* hmr(void *ptr)
            cores to become ready and starts all cores with a write to the
            shared memory location */
 
-        /* Set affinity to core 0 */
-        CPU_SET(0, &affin_mask);
-        sched_setaffinity(0, sizeof(cpu_set_t), &affin_mask);
+        if (!safemode) {
+            /* Set affinity to core 0 */
+            CPU_SET(0, &affin_mask);
+            sched_setaffinity(0, sizeof(cpu_set_t), &affin_mask);
+        }
 
         /* Spin until the appropriate numer of threads have become ready */
         wait64(&ready_lock, nthrds - 1);
         clock_gettime(CLOCK_MONOTONIC, &tv_monot_start);
         fetchadd64_release(&sync_lock, 1);
     }
     else {
-        /* Calculate affinity mask for my core and set affinity */
-        CPU_SET(((mycore >> 1)) + ((ncores >> 1) * (mycore & 1)), &affin_mask);
-        sched_setaffinity(0, sizeof(cpu_set_t), &affin_mask);
+        if (!safemode) {
```

Comment:
We probably need to have sched_setaffinity even in safemode. Because sched_setaffinity doesn't need root permission.